lib/rewards/server/subscribers/rewards-subscriber.js

```diff
@@ -0,0 +1,135 @@
+const { sc, Logger } = require('@reldens/utils');
+
+class RewardsSubscriber
+{
+    static async onBattleEnded(playerSchema, targetObject, events)
+    {
+        let rewards = sc.get(targetObject, 'rewards', false);
+
+        let winningRewards = RewardsSubscriber.getWinningRewards(rewards, false);
+        if (0 === winningRewards.length) {
+            return false;
+        }
+        let itemRepository =  targetObject.dataServer.getEntity('item')
+
+        for (let winningReward of winningRewards) {
+            let winningItem = await itemRepository.loadById(winningReward.item_id);
+            let addItemsResult = await RewardsSubscriber.addItemToPlayerInventory(playerSchema, winningItem, winningReward.drop_quantity);
+            if (!addItemsResult) {
+                Logger.error('Couldn\'t add item to Player\'s Inventory');
+                return false;
+            }
+
+            await events.emit('reldens.rewardDropped', {
+                reward: winningReward,
+                item_name: winningItem.label,
+                player_id: playerSchema.player_id,
+                room_id: targetObject.room_id,
+                client: playerSchema?.skillsServer?.client?.client
+            });
+        }
+        await RewardsSubscriber.updateUniqueRewards(winningRewards, targetObject.dataServer);
+        return true;
+    }
+
+    static async addItemToPlayerInventory(playerSchema, item, quantity)
+    {
+        let rewardItemInstance = playerSchema.inventory.manager.createItemInstance(item.key);
+        let rewardItems = Array(quantity).fill(rewardItemInstance);
+        return await playerSchema.inventory.manager.addItems(rewardItems);
+    }
+
+    static getWinningRewards(rewards, usesRewardBag)
+    {
+        let rewardAwarded = [];
+        if (!RewardsSubscriber.areValidRewards(rewards)) {
+            return rewardAwarded;
+        }
+        if (usesRewardBag) {
+            let rewardsBag = RewardsSubscriber.getRewardsBag(rewards);
+            let reward = RewardsSubscriber.pickReward(rewardsBag);
+            if (RewardsSubscriber.isValidReward(reward)) {
+                rewardAwarded.push(reward);
+            }
+            return rewardAwarded;
+        }
+        for (let reward of rewards) {
+            if (RewardsSubscriber.isRewardAwarded(reward)) {
```

Comment:
avoid nested IFs, use the opposite to continue with the loop:
```
if(!this.isRewardAwarded(reward)){
    continue;
}
```

Reply:
and that way you can do both at once:
```
if(!this.isRewardAwarded(reward) || !this.isValidReward(reward)){
    continue;
}
```

Reply:
yeap! done!